### ISSUE TITLE
white screen

### ISSUE THREAD

**Issue by Iexhu** [ififfy/flipflip#211]
Hi StarrHelixx,

Coming back on the issue of the white screen "freeze", according to the log files i get the following two errors although they are identical. It doesnt matter if i play vids or pics or a combo, after a while i get the white screen.

index.html:1 Uncaught (in promise) DOMException: The play() request was interrupted by a call to pause(). https://goo.gl/LdLk22
index.html:1 Uncaught (in promise) DOMException: The play() request was interrupted by a call to pause(). https://goo.gl/LdLk22

**Comment by StarrHelixx** [comment]
I _just_ experienced this for the first time today. But wasn't able to reliably recreate the problem... so still unsure of the cause. When it happened to me, I opeend the Dev Tools, and it said something like "DevTools was disconnected from the page" is that what you saw too?

I'm assuming the errors you pasted above were what you saw _after_ you reloaded. This is actually a 'feature' of Chrome and happens when you are reloading and a video is played. Chrome introduced this to prevent videos from autoplaying on a page a user hasn't clicked on yet.

Anyways, I will continue to try and reproduce this issue, but it smells like an inner problem with Electron and not FlipFlip itself.

**Comment by Iexhu** [comment]
Will have to take a closer look. Did you follow the link from the error message? There is a whole webpage discussing the same error which could be helpful to you.

**Comment by StarrHelixx** [comment]
Yes, but that error is a red herring. That will happen occasionally while running FlipFlip, but doesn't cause the app the crash.

**Comment by Iexhu** [comment]
Sorry for the one and a half week gap between responses. I have been trying to reproduce the issue in various settings, so far no white screen freeze yet. I remember there being a win10 update recently, maybe that solved it. If it ever reoccurs i will let you know.

**Comment by StarrHelixx** [comment]
No problem lexhu, thanks for the update. I've had this occur once or twice to me recently, I believe it happens when FlipFlip runs out of memory. At this point, the overhead for the UI has gotten quite large, but a majority of this is due to the Electron framework and not specifically FlipFlip code, making it difficult to optimize/improve. That being said, going to try and do some additional work to reduce/improve memory use for the next release.

**Comment by Iexhu** [comment]
To be honest i was thinking it did feel like a memory issue but speculation is not very helpfull, so i did not mention it. Adding more to cache might help? Again i'm not a coder and can draw only from my experience as a pc user. Your efforts are much appreciated!

**Comment by StarrHelixx** [comment]
So... I've been really trying to get to the bottom of FlipFlip's memory issues, but don't feel like I've made a ton of progress. I do feel even _more_ confident that it is simply Electron being an outright memory-hog and not really issues with FlipFlip's codebase.

To start, I have had reports of FlipFlip just playing the same images over and over after playing for awhile. I experienced this myself the other day. This seems to be caused by the fetch loops improperly ending. I, unfortunately, wasn't able to reliably reproduce this, but I'm continuing my tests on this.

However, while doing these tests, I did notice that memory just skyrockets to like 15GB. That said, I _was_ able to validate that this is just Electron using any _available_ memory -- not necessarily a bad thing. Free memory is consumed until it is gone, at which point garbage is collected to free up additional memory -- and this continues to happen without interrupting FlipFlip playback. That said, FlipFlip does seem to slow down after like 30 minutes and I'm still trying to nail down the cause of that.

tldr; Don't worry _too_ much about FlipFlip consuming lots of memory, this is expected. But whitescreens/slow-downs are more of an issue

**Comment by Iexhu** [comment]
Had a couple more white screens. Never was i able to find any more errors even if i was fast enough to pause after closing and reopening. The white screen can only be closed and not be interacted with so thats not very helpfull either. 

Concerning memory;
What i noticed over the last weeks is that it happens sooner when you use multiple sources and playing at faster speeds. Also the frequency at which flipflip goes white screen increases just using the same sources and setup after closing and reopening. Does that mean that the memory is not getting purged after closing? so it keeps filling up and therefore leads to it crashing sooner? 

I use 8GB of memory so it will crash sooner i suppose. Not much usefull information but there you have it.

**Comment by Iexhu** [comment]
Hi again StarrHelixx,

This error popped up recently. I was playing a youtube video in the background whilst looking at a slideshow when both youtube and the app crashed simultanously. For clarity the app went to a white screen. 

[-1584909272584.log](https://github.com/ififfy/flipflip/files/4382846/-1584909272584.log)

**Comment by katzlbt** [comment]
I do not know how often fliflip saves this file. It seems to be multiple times a second or so. This should not be necessary and drains SSD lifetime. By typing in a color-code manually while playing you get an unrecoverable white screen crash that is saved in the data.json file and leads to a white-screen on restart.

data.json:
         "strobeColor":"#f",

**Comment by StarrHelixx** [comment]
@katzlbt Yea, FlipFlip just runs a loop which saves once every [500ms](https://github.com/ififfy/flipflip/blob/master/src/renderer/components/Meta.tsx#L53), which is totally unnecessary, but is a hold-over from old code. This could be improved to only save when a change happens. 

As for the strobe issue, this is separate from this issue ticket, but has now been resolved on the `master` branch.

**Comment by StarrHelixx** [comment]
FlipFlip now saves much more intelligently, and rolling back Electron seems to have resolve this issue. Closing for now
